### PR TITLE
#515 css only packages

### PR DESCRIPTION
**Pull request by mjcarsjens**
Added a CSS-only package. This contains 3 files:
- index.css (component css + DH design tokens)
- bem.css (component css only)
- root-theme.css (DH design tokens only)

Also adds `extract: true` to `postcss` plugin in `rollup.config.js` since we need the `.css` files here, but this also solves the first part of #492 .

**Closing issues**
#515

### CODE REVIEW

**Comment by mjcarsjens** [comment]
@bddjong Fixed the broken visual regression tests. Some context:

Adding `extract: true` to the `postcss` plugin in rollup seems to break some styling for a few components, resulting in the failed visual regression tests. For now I've removed the `extract: true` from our rollup config, duplicated the rollup config to the css package folder and added `extract: true` to that specific config. Once we've worked out why the styling breaks for these specific components and we've re-added the `extract: true` to the global rollup config we can remove the specific one and point to the global file again.

For now fixing this seems like out of scope for this specific issue though, which is why I went with this approach.

**Comment by Robbert** [comment]
I was thinking of renaming `bem.css` into `index.css`. I think it makes sense, given the design token architecture, the tokens are not bundled by default.

```js
import '@denhaag/component-library';
import '@utrecht/design-tokens';
```

This code above looks like it should work, but would give weird results because there are now two sets of design tokens in the page. The following code could be the "fix", but I think I would rather make the components without design tokens the default instead:

```js
import '@denhaag/component-library/dist/bem';
import '@utrecht/design-tokens';
```

@mjcarsjens What do you think?

**Comment by mjcarsjens** [comment]
> I was thinking of renaming `bem.css` into `index.css`. I think it makes sense, given the design token architecture, the tokens are not bundled by default.
> 
> ```js
> import '@denhaag/component-library';
> import '@utrecht/design-tokens';
> ```
> 
> This code above looks like it should work, but would give weird results because there are now two sets of design tokens in the page. The following code could be the "fix", but I think I would rather make the components without design tokens the default instead:
> 
> ```js
> import '@denhaag/component-library/dist/bem';
> import '@utrecht/design-tokens';
> ```
> 
> @mjcarsjens What do you think?

@Robbert I think that's a fair point, it indeed would make it more intuitive to use when the design tokens are not by default bundled in the `index.css`. Do we want to include the design-tokens at all in this package (as a design-tokens only package already exists) or just leave them seperated? I feel like maybe this package should not include the design-tokens at all and just leave it up to the user to decide which token package to use.

**Comment by mjcarsjens** [comment]
@Robbert I just made a commit of what it would look like with only the components css and no design-tokens in this package.

**Comment by Robbert** [comment]
@mjcarsjens Looks good! Only thing I'm wondering is if having `dist/index.css` is enough, or if you actively want to provide and maintain the `.js` and `.d.ts` files as well. If they are just a side effect from using rollup, you could consider putting `*.js` and `*.ts` in `.npmignore`.

**Comment by mjcarsjens** [comment]
@Robbert you're right, in the previous version it indeed did only build 3 `.css` files but seems like I got a bit over-enthusiastic with the rewriting. Changed the build process so now only a single  `.css` file appears in the `dist` folder. Tested it in another `tsx` file and as long as the `main` is set to the `dist/index.css` file in `package.json` everything works as intended.

**Comment by mjcarsjens** [comment]
@bddjong any thoughts?

File: packages/component-library-css/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@gemeente-denhaag/component-library-css",
+  "version": "0.2.3-alpha.50",
```
Comment (bddjong) [inline packages/component-library-css/package.json:3]: I'm guessing this package isn't starting on alpha version 50 ;o

File: packages/component-library-css/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "A CSS library for the components from the Municipality of the Hague",
+  "bugs": "https://github.com/nl-design-system/denhaag/issues",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/nl-design-system/denhaag.git",
+    "directory": "src/"
```
Comment (bddjong) [inline packages/component-library-css/package.json:9]:
```suggestion
    "directory": "packages/component-library-css"
```

File: packages/component-library-css/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  '@material-ui/core',
+  '@material-ui/lab',
+  '@material-ui/pickers',
+  'react',
+  'react-dom',
+  '@material-ui/core/transitions',
+  '@material-ui/core/OverridableComponent',
+];
+const internalDependencies = _.keys(tsconfig.compilerOptions.paths);
+const dependencies = externalDependencies.concat(internalDependencies);
```
Comment (bddjong) [inline packages/component-library-css/rollup.config.js:21]: Do we actually need a separate rollup config for this package? In any case, I believe there shouldn't be need for the external code if we're only exporting css from this package.

**Review by bddjong — CHANGES_REQUESTED** [review]

File: packages/component-library-css/package.json (hunk 1 above)
Comment (Robbert) [reply packages/component-library-css/package.json:3]: I make the same mistake all the time 🙈

File: packages/component-library-css/rollup.config.js (hunk 3 above)
Comment (mjcarsjens) [reply packages/component-library-css/rollup.config.js:21]: > @bddjong Fixed the broken visual regression tests. Some context:
> 
> Adding `extract: true` to the `postcss` plugin in rollup seems to break some styling for a few components, resulting in the failed visual regression tests. For now I've removed the `extract: true` from our rollup config, duplicated the rollup config to the css package folder and added `extract: true` to that specific config. Once we've worked out why the styling breaks for these specific components and we've re-added the `extract: true` to the global rollup config we can remove the specific one and point to the global file again.
> 
> For now fixing this seems like out of scope for this specific issue though, which is why I went with this approach.

This comment explains the need for the seperate rollup config, probably could indeed use some changes but as it's not intended to be used long-term not sure whether this is worth our time.

**Review by bddjong — APPROVED** [review]

File: packages/component-library-css/package.json (hunk 1 above)
Comment (mjcarsjens) [reply packages/component-library-css/package.json]: Whoops, is fixed now

**Review by bddjong — APPROVED** [review]

**Pull request merged**